packages/views.py

```diff
@@ -0,0 +1,281 @@
+from django.core import serializers
+from django.shortcuts import render, redirect
+from .models import Order, City, Address
+from .models import Partner
+from .models import Package
+from util import is_member, is_staff
+
+
+def partner_search(request, partner_name):
+    partner = Partner.objects.filter(name=partner_name).get()
+
+    name_query = request.GET.get('name')
+    street_name_query = request.GET.get('street_name')
+    street_number_query = request.GET.get('street_number')
+    city_query = request.GET.get('city')
+    number_query = request.GET.get('number')
+
+    packages = Package.objects.filter(order__partner__name=partner.name)
+    filtered_packages = Package.objects.none()
+
+    if name_query:
+        packages = packages.filter(full_name__icontains=name_query)
+        filtered_packages = packages
+    if number_query:
+        packages = packages.filter(phone_number__icontains=number_query)
+        filtered_packages = packages
+    if street_name_query:
+        packages = packages.filter(destination__street_name__icontains=street_name_query)
+        filtered_packages = packages
+    if street_number_query:
+        packages = packages.filter(destination__street_number__contains=street_number_query)
+        filtered_packages = packages
+    if city_query:
+        packages = packages.filter(destination__city__name__icontains=city_query)
+        filtered_packages = packages
+
+    orders = []
+    related_packages = []
+    for package in filtered_packages.order_by('-order__collection_date'):
+        order = package.order
+        if order not in orders:
+            orders.append(order)
+            related_packages.append([package])
+        else:
+            related_packages[orders.index(order)].append(package)
+
+    order_packages = [(orders[i], related_packages[i]) for i in range(0, len(orders))]
+    context = {
+        'partner': partner,
+        'order_packages': order_packages,
+    }
+    context.update(string_data_lists_context())
+
+    return render(request, 'packages/partner_search.html', context)
+
+
+def order_view(request, order_id):
+    order = Order.objects.get(id=order_id)
+
+    if request.POST:
+        if request.POST.get('package') == 'all':
+            for inner in order.related_packages():
+                inner.as_query().update(status=inner.next_status())
+        else:
+            package_id = request.POST.get('package')
+            package = order.related_packages().filter(id=package_id)
+            package.update(status=package.get().next_status())
+
+    packages = list(order.related_packages())
+    context = {
+        'order': order,
+        'packages': packages,
+        'is_staff': is_staff(request.user)
+    }
+
+    return render(request, "packages/order.html", context)
+
+
+def package_view(request, package_id):
+    package = Package.objects.get(id=package_id)
+    name = package.full_name.split(' ')[0] + ':'
+
+    if not name:

```

Comment:
How could this be possible?

Reply:
If the name wasn't specified, full_name has blank=True

Reply:
If the name is empty, since full_name could be blank, it's possible

Reply:
So you mean it's empty? That will never happen because you do `+ ':'` above it.

Reply:
just removed this and replaced the `:` with a ' '